libfwupdplugin/fu-bios-settings.c

```diff
@@ -440,6 +511,39 @@ fu_bios_settings_setup(FuBiosSettings *self, GError **error)
 				return FALSE;
 			}
 		} while (++count);
+
+		/* read all authentication info */
+		authentication_path = g_build_filename(sysfsfwdir, driver, "authentication", NULL);
+		if (!g_file_test(authentication_path, G_FILE_TEST_IS_DIR)) {
+			g_debug("skipping non-directory %s", authentication_path);
+			continue;
+		}
+		driver_auth_dir = g_dir_open(authentication_path, 0, error);
+		if (driver_auth_dir == NULL)
+			return FALSE;
+		do {
+			const gchar *name = g_dir_read_name(driver_auth_dir);
+			g_autofree gchar *full_path = NULL;
+			g_autoptr(GError) error_local = NULL;
+			if (name == NULL)
+				break;
+			full_path = g_build_filename(authentication_path, name, NULL);
+			if (!fu_bios_settings_populate_authentication(self,
```

Comment:
is the logic here that you can have two "active" authentication mechanisms, e.g. "password" **OR** "certificate"?